src/descrambler/cccam.c

```diff
@@ -1851,17 +1851,24 @@ caclient_cccam_nodeid_set(void *o, const void *v)
   cccam_t *cccam = o;
   const char *s = v ?: "";
   uint8_t key[8];
-  int i, u, l;
-
-  for(i = 0; i < ARRAY_SIZE(key); i++) {
-    while(*s != 0 && !isxdigit(*s)) s++;
-    u = *s ? nibble(*s++) : 0;
-    while(*s != 0 && !isxdigit(*s)) s++;
-    l = *s ? nibble(*s++) : 0;
-    key[7-i] = (u << 4) | l;
+  int u, l, i = 0;
+
+  if (!strlen(v)) {
```

Comment:
modified: now comparing result with zero
Also fixed a bug on the byte display order (endianess)